### ISSUE TITLE
Slow Performance on IE with Leaflet

### ISSUE THREAD

**Issue by jaxoncreed** [angular/material#7540]
When running an Angular app with Angular-Material and [Leaflet](http://leafletjs.com/) performance significantly decreases in Internet Explorer.

I've traced the issue to the mdThemingProvider:

/material/src/core/core.js
```
  $mdThemingProvider.theme('default')
    .primaryPalette('indigo')
    .accentPalette('pink')
    .warnPalette('deep-orange')
    .backgroundPalette('grey');
```

Removing these 5 lines will increase the performance in IE9 from 15fps to 30fps.

I'm hoping for some guidance on what's happening and what I should be looking into in order to find a fix.

Thanks

**Comment by 0101adm** [comment]
+1

 im also seeing perf decrease with google-maps in Chrome.

i'll have to test what youve done.

**Comment by jaxoncreed** [comment]
For more information on this issue, I've found three things that effect the FPS in this specific app, two of which are in angular material:

```
+--------------+---------+-----------------+-----+
| core.theming | styling | Angular Leaflet | FPS |
+--------------+---------+-----------------+-----+
|            0 |       0 |               0 |  30 |
|            0 |       0 |               1 |  12 |
|            0 |       1 |               0 |   5 |
|            1 |       0 |               0 |  15 |
|            1 |       1 |               0 |  10 |
|            1 |       1 |               1 |   4 |
+--------------+---------+-----------------+-----+
```

I'll continue to strip out pieces of the styling and theming to continue to diagnose the issue and keep updates posted here.

**Comment by jaxoncreed** [comment]
I've created a simple repo for anyone hoping to do some testing on this here (https://github.com/jaxoncreed/Leaflet-AngularMaterial-IE-Test).

**Comment by crisbeto** [comment]
I've also been trying to reproduce this. It really is slow on IE, but I didn't manage to make it better by disabling those styles.

**Comment by jaxoncreed** [comment]
Pretty sure this is most of the issue is caused by attribute selectors. Apparently IE will check every single element that has an attribute for every element selector in the stylesheet.

There are a large number of element selectors between lines 5170, 14085 in angular-material.css. The more of these you remove, the better performance gets.

**Comment by topherfangio** [comment]
Is anyone seeing this in HEAD? I noticed some CSS that got added to 1.0.6 that may not have been necessary, but I am still trying to confirm. If removed, the performance is back where it should be. And I'm not seeing this CSS in master.

**Comment by crisbeto** [comment]
Can confirm that the performance is fine when running master.

**Comment by ThomasBurleson** [comment]
@topherfangio - what CSS sections are you referring to (in 1.0.6)?

**Comment by topherfangio** [comment]
Just to update the thread since Thomas and I talked offline, we accidentally added some attribute selector code into the 1.0.6 release which was intended to be in a separate/optional file. It has been fixed in master so it should not cause an issue in 1.1.0.

So, that should resolve half of this issue. It looks like we still need to dig in and find out why the themes are causing an issue.

**Comment by david-gang** [comment]
Hi all,

When i set app.constant('$MD_THEME_CSS', ''); the performance is much better but than i am without theming :-)

Even afterwards the performance in ie is not too great, so i think that there are additional css bottlenecks in the non theming css as well.

Thanks,
David

**Comment by david-gang** [comment]
Hi @topherfangio ,
Is the fix of the attribute selectors also included in the version 1.0.7?
If yes, then it is not the whole story, as it works much slower than 1.0.5. 

Thanks,
David

**Comment by topherfangio** [comment]
@david-gang Yes, the fix for the attribute selectors are included in 1.0.7. I'd love to find out what changed between 1.0.5 and 1.0.7 that could be affecting performance.

Do you have a way to checkout specific versions of Material and use them in your app?

**Comment by ThomasBurleson** [comment]
This issue is closed as part of our ‘Surge Focus on Material 2' efforts.
      For details, see our forum posting @ http://bit.ly/1UhZyWs.
